enhancements/rhcos/rhcos-inject.md

```diff
@@ -0,0 +1,198 @@
+
+An RHCOS node is deployed into an environment without DHCP and needs to be explicitly configured with an IP address/mask, gateway, and hostname:
+
+```console
+$ rhcos-inject \
+	--input=rhcos-installer.x86_64.iso \
+	--output=control-0.iso \
+	--openshift-config=master.ign \
+	--ip=10.10.10.2::10.10.10.254:255.255.255.0:control0.example.com:enp1s0:none
+```
+
+This static IP configuration is used for both the RHCOS installation and the post-installation system.
+
+#### Dedicated Provisioning Network
+
+An RHCOS node is deployed into an environment which makes use of a dedicated provisioning network which is only used during installation:
+
+```console
+$ rhcos-inject \
+	--input=rhcos-installer.x86_64.iso \
+	--output=control-0.iso \
+	--openshift-config=master.ign \
+	--ip=10.10.10.2::10.10.10.254:255.255.255.0:control0.example.com:enp1s0:none \
+	--ip=:::::enp1s1:none \
+	--persist-network=false
+```
+
+This static IP configuration is used during the installation, but once the machine reboots into the running system, it uses a different configuration. This is likely to be paired with the following use case.
+
+#### Custom Dynamic Host Configuration
+
+An RHCOS node is deployed into an environment which uses an in-house IPAM implementation in lieu of DHCP:
+
+```console
+$ cat network.ign
+{
+  "ignition": { "version": "3.0.0"},
+  "systemd": {
+    "units": [{
+      "name": "configure-networking.service",
+      "enabled": true,
+      "contents": "[Service]\nType=oneshot\nExecStart=/usr/bin/env nmcli ...\n\n[Install]\nWantedBy=pre-install.target"
+    }]
+  }
+}
+
+$ rhcos-inject \
+	--input=rhcos-installer.x86_64.iso \
+	--output=my-rhcos-installer.x86_64.iso \
+	--openshift-config=master.ign \
+	--installer-config=network.ign
+```
+
+This service can contain just about any piece of logic needed in order to statically configure the node based on a dynamic assignment. For example, a customer may use this mechanism to configure a link-local address, request an IP from a provisioning system, reconfigure the network interfaces, and then phone-home to acknowledge a successful configuration.
+
+### Implementation Details/Notes/Constraints
+
+As with all software, it's important to consider coupling and cohesion when looking at this solution and its alternatives. Defining clear API boundaries and building upon layers of abstraction are some of the most effective techniques for avoiding pitfalls. This solution chooses to make a distinction between the networking required for an individual node to operate and for the cluster itself; respectively, the machine network and the pod network. A functioning machine network is considered a prerequisite to installation, as is power, cooling, and many others. The pod network, on the other hand, is something created and managed by OpenShift. Working backward from this assumption, it's clear that the solution to the problem of pre-installation network configuration should not be solved by the cluster.
+
```

Comment:
Network configuration isn't just an installation time problem - there may be changes over time which impact the configuration created by this workflow - how do we envisage customers dealing with that?

It seems like the answer is probably ssh/scp or ansible - and that's quite inconsistent with the user experience for everything else (including secondary nics which aren't needed during initial deployment.